middleware/markdown/metadata.go

```diff
@@ -183,43 +178,29 @@ func (y *YAMLMetadataParser) Closing() []byte {
 // It returns the metadata, the remaining bytes (markdown), and an error, if any.
 func extractMetadata(parser MetadataParser, b []byte) (metadata []byte, markdown []byte, err error) {
 	b = bytes.TrimSpace(b)
-	reader := bufio.NewReader(bytes.NewBuffer(b))
-
-	// Read first line, which should indicate metadata or not
-	line, err := reader.ReadBytes('\n')
-	if err != nil || !bytes.Equal(bytes.TrimSpace(line), parser.Opening()) {
+	openingLine := append(parser.Opening(), '\n')
+	ClosingLine := append(parser.Closing(), '\n')
+	if !bytes.HasPrefix(b, openingLine) {
 		return nil, b, fmt.Errorf("first line missing expected metadata identifier")
 	}
-
-	// buffer for metadata contents
-	metaBuf := bytes.Buffer{}
-
-	// Read remaining lines until closing identifier is found
-	for {
-		line, err := reader.ReadBytes('\n')
-		if err != nil && err != io.EOF {
-			return nil, nil, err
-		}
-
-		// if closing identifier found, the remaining bytes must be markdown content
-		if bytes.Equal(bytes.TrimSpace(line), parser.Closing()) {
-			break
-		}
-
-		// if file ended, by this point no closing identifier was found
-		if err == io.EOF {
-			return nil, nil, fmt.Errorf("metadata not closed ('%s' not found)", parser.Closing())
-		}
-
-		metaBuf.Write(line)
-		metaBuf.WriteString("\r\n")
+	metaStart := len(openingLine)
+	if _, ok := parser.(*JSONMetadataParser); ok {
```

Comment:
You're right. I tried avoiding this too.

Initially we had nested objects, which is why we went with the parser and could not go this route. i.e. you can't just search for closing identifier because there can be many of them. Luckily, its simplified now.

How about manually adding the opening `{` and closing `}` braces before passing it to json decoder inside `Parse()` of `JSONMetadataParser` ?

Finally, we need to hint users not to do this 
```
{ "title" : "Some title"
... }
```
Though its a valid json, the variables must start at the next line to be recognised.


Thanks :+1: